lib/codecs/src/decoding/format/gelf.rs

```diff
@@ -0,0 +1,396 @@
+pub mod gelf_fields {
+
+    /// <not a field> The latest version of the GELF specificaiton.
+    pub const GELF_VERSION: &str = "1.1";
+
+    /// (required) GELF spec version
+    pub const VERSION: &str = "version";
+
+    /// (required) The name of the host, source or application that sent this message.
+    pub const HOST: &str = "host";
+
+    /// (required) A short descriptive message.
+    pub const SHORT_MESSAGE: &str = "short_message";
+
+    /// (optional) A long message that can i.e. contain a backtrace
+    pub const FULL_MESSAGE: &str = "full_message";
+
+    /// (optional) Seconds since UNIX epoch with optional decimal places for milliseconds.
+    ///  SHOULD be set by client library. Will be set to the current timestamp (now) by the server if absent.
+    pub const TIMESTAMP: &str = "timestamp";
+
+    /// (optional) The level equal to the standard syslog levels. default is 1 (ALERT).
+    pub const LEVEL: &str = "level";
+
+    /// (optional) (deprecated) Send as additional field instead.
+    pub const FACILITY: &str = "facility";
+
+    /// (optional) (deprecated) The line in a file that caused the error (decimal). Send as additional field instead.
+    pub const LINE: &str = "line";
+
+    /// (optional) (deprecated) The file (with path if you want) that caused the error. Send as additional field instead.
+    pub const FILE: &str = "file";
+}
+pub use gelf_fields::*;
+
+/// Config used to build a `GelfDeserializer`.
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
+pub struct GelfDeserializerConfig;
+
+impl GelfDeserializerConfig {
+    /// Build the `GelfDeserializer` from this configuration.
+    pub fn build(&self) -> GelfDeserializer {
+        GelfDeserializer::default()
+    }
+
+    /// Return the type of event built by this deserializer.
+    pub fn output_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    /// The schema produced by the deserializer.
+    pub fn schema_definition(&self) -> schema::Definition {
+        schema::Definition::empty()
+            .with_field(VERSION, Kind::bytes(), None)
+            .with_field(HOST, Kind::bytes(), None)
+            .with_field(SHORT_MESSAGE, Kind::bytes(), None)
+            .optional_field(FULL_MESSAGE, Kind::bytes(), None)
+            .optional_field(TIMESTAMP, Kind::timestamp(), None)
+            .optional_field(LEVEL, Kind::integer(), None)
+            .optional_field(FACILITY, Kind::bytes(), None)
+            .optional_field(LINE, Kind::integer(), None)
+            .optional_field(FILE, Kind::bytes(), None)
+            // Every field with an underscore (_) prefix will be treated as an additional field.
+            // Allowed characters in field names are any word character (letter, number, underscore), dashes and dots.
+            // Libraries SHOULD not allow to send id as additional field ( _id). Graylog server nodes omit this field automatically.
+            .unknown_fields(Kind::bytes())
```

Comment:
Looking at https://docs.graylog.org/docs/gelf I think unknown fields can be strings or numbers.

Reply:
Unfortunately I'm not seeing a`value::Kind` type that covers more than one type :/
I had set it to bytes after discussing with Pablo.

Perhaps a comment explaining that would help?
Otherwise I can look into what it would entail to change the infrastructure code to allow multiple types to be set, or something like that.

Reply:
Ah, yeah, hrm, I do see that. I am curious how much work it would be to expand this to allow multiple types because, as it is, this will provide incorrect type information to VRL (it'll think all extra fields are a string and then panic if it isn't). If anything, I'd like to make this `any()` which isn't as strict as it could be, but would at least avoid panics due to mismatched type definitions.

cc/ @JeanMertz @fuchsnj for thoughts on expanding the schema to allow union types here.

Reply:
ah, I did not realize `any()` was an option- I missed that Kind is a struct and not an Enum.

Reply:
Should we create a github issue to investigate union types ?

Reply:
Yeah, a separate issue makes sense to me.

Reply:
There are builder methods to make a "union" type. `Kind::bytes().or_integer().or_float()` might be what you were looking for

Reply:
> There are builder methods to make a "union" type. `Kind::bytes().or_integer().or_float()` might be what you were looking for

I think that's exactly what we were looking for!